src/pkg/scan/scanner/api/registration.go

```diff
@@ -0,0 +1,194 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"github.com/goharbor/harbor/src/core/promgr/metamgr"
+	"github.com/goharbor/harbor/src/jobservice/logger"
+	"github.com/goharbor/harbor/src/pkg/q"
+	rscanner "github.com/goharbor/harbor/src/pkg/scan/scanner"
+	"github.com/goharbor/harbor/src/pkg/scan/scanner/dao/scanner"
+	"github.com/goharbor/harbor/src/pkg/scan/scanner/scan"
+	"github.com/pkg/errors"
+)
+
+const (
+	proScannerMetaKey = "projectScanner"
+)
+
+// DefaultController is a singleton api controller for plug scanners
+var DefaultController = New()
+
+// New a basic controller
+func New() Controller {
+	return &basicController{
+		manager:    rscanner.New(),
+		proMetaMgr: metamgr.NewDefaultProjectMetadataManager(),
+	}
+}
+
+// basicController is default implementation of api.Controller interface
+type basicController struct {
+	// managers for managing the scanner registrations
+	manager rscanner.Manager
+	// for operating the project level configured scanner
+	proMetaMgr metamgr.ProjectMetadataManager
+	// controller for scan actions
+	c scan.Controller
+	// Client
+}
+
+// ListRegistrations ...
+func (bc *basicController) ListRegistrations(query *q.Query) ([]*scanner.Registration, error) {
+	return bc.manager.List(query)
+}
+
+// CreateRegistration ...
+func (bc *basicController) CreateRegistration(registration *scanner.Registration) (string, error) {
+	// TODO: Get metadata from the adapter service first
+	l, err := bc.manager.List(nil)
```

Comment:
This call will be very expensive when there are many registrations in the system, suggest to list the registrations limit to one.

Reply:
Actually, the registrations will not be on a large scale "state". Anyway, I'll take care of it. As it's a perf issue, if no other obvious issues, I'll change it in the later PR.

Reply:
```go
// Check if there is any registrations already existing.
	l, err := bc.manager.List(&q.Query{
		PageSize:   1,
		PageNumber: 1,
	})
	if err != nil {
		return "", err
	}```